webapp/TargetedMS/js/QCPlotHelperWrapper.js

```diff
@@ -7,70 +7,151 @@ Ext4.define("LABKEY.targetedms.QCPlotHelperWrapper", {
     mixins: {
         leveyJennings: 'LABKEY.targetedms.LeveyJenningsPlotHelper',
         cusum: 'LABKEY.targetedms.CUSUMPlotHelper',
-        movingRange: 'LABKEY.targetedms.MovingRangePlotHelper'
+        movingRange: 'LABKEY.targetedms.MovingRangePlotHelper',
+        trailingMean: 'LABKEY.targetedms.TrailingMeanPlotHelper',
+        trailingCV: 'LABKEY.targetedms.TrailingCVPlotHelper'
     },
 
     statics: {
-        getQCPlotTypeLabel: function(visPlotType, isCUSUMMean)
-        {
-            if (visPlotType == LABKEY.vis.TrendingLinePlotType.MovingRange)
+        getQCPlotTypeLabel: function(visPlotType, isCUSUMMean) {
+            if (visPlotType === LABKEY.vis.TrendingLinePlotType.MovingRange)
                 return 'Moving Range';
-            else if (visPlotType == LABKEY.vis.TrendingLinePlotType.CUSUM)
-            {
+            else if (visPlotType === LABKEY.vis.TrendingLinePlotType.CUSUM) {
                 if (isCUSUMMean)
                     return 'CUSUMm';
                 else
                     return 'CUSUMv';
             }
+            else if (visPlotType === LABKEY.vis.TrendingLinePlotType.TrailingMean) {
+                return 'Trailing Mean';
+            }
+            else if (visPlotType === LABKEY.vis.TrendingLinePlotType.TrailingCV) {
+                return 'Trailing CV';
+            }
             else
                 return 'Levey-Jennings';
         }
     },
 
-    addIndividualPrecursorPlots : function()
-    {
+    deepCloneArray: function (arr) {
+        return arr.map(function(item) {
+            // Check if the item is an array
+            if (Array.isArray(item)) {
+                // If it is, call deepCloneArray on it
+                return this.deepCloneArray(item);
+            } else if (typeof item === 'object' && item !== null) {
+                // If it's an object, use Object.assign to create a new object with the same properties
+                return Object.assign({}, item);
```

Comment:
I would have expected this to call deepCloneObject(). Otherwise it won't do a deep clone - it'll share the same objects as values, right?